src/components/NavBar/Profile.tsx

```diff
@@ -0,0 +1,51 @@
+import { useReactOidc } from '@axa-fr/react-oidc-context';
+import React from 'react';
+import {
+    DropdownItem,
+    DropdownMenu,
+    DropdownToggle,
+    UncontrolledDropdown
+} from 'reactstrap';
+
+interface ProfileProps {}
+
+const Profile: React.FunctionComponent<ProfileProps> = () => {
+    const { oidcUser, logout, events } = useReactOidc();
+
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+    const toggle = () => {
+        setIsOpen(!isOpen);
+    };
+
+    if (!oidcUser) return null;
+
+    const {
+        profile: { name, preferred_username }
+    } = oidcUser;
+
+    return (
+        <UncontrolledDropdown nav inNavbar>
+            <DropdownToggle nav caret className="navbar-user">
+                <img
+                    className="rounded-circle"
+                    src={`https://profiles.csh.rit.edu/image/${preferred_username}`}
+                    alt=""
+                    aria-hidden={true}
+                    width={32}
+                    height={32}
+                />
+                {name} ({preferred_username})
+                <span className="caret" />
+            </DropdownToggle>
+            <DropdownMenu>
+                <DropdownItem>Dashboard</DropdownItem>
+                <DropdownItem>Settings</DropdownItem>
+                <DropdownItem divider />
+                <DropdownItem onClick={() => logout()}>Logout</DropdownItem>
```

Comment:
No need for the extra function here, just directly pass `logout`:
```suggestion
                <DropdownItem onClick={logout}>Logout</DropdownItem>
```

Reply:
I'm afraid we can't. The typing returned clashes in the ts compiler